### PR TITLE
[DNM]refactor(gaussdb): move gaussdb resources to services

### PR DESCRIPTION
**Pull request by niuzhenguo**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccSomethingV0_basic'
...
=== RUN   TestAccSomethingV0_basic
--- PASS: TestAccSomethingV0_basic (70.75s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       70.796s
```

### CODE REVIEW

**Comment by chengxiangdong** [comment]
`Providers` is deprecated, it is better to use `ProviderFactories` in the unit test files

File: huaweicloud/services/gaussdb/transport.go [hunk 1]
```diff
@@ -0,0 +1,207 @@
+package gaussdb
```
Comment (chengxiangdong) [inline huaweicloud/services/gaussdb/transport.go:1]: I see that some functions are used in some other resources or data sources in huaweicloud pakcage, so suggest put the `transport.go` file the common path.

File: huaweicloud/services/gaussdb/transport.go [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+	return stateConf.WaitForState()
+}
+
+func checkCsClusterV1DeleteFinished(data interface{}) bool {
+	c, err := navigateValue(data, []string{"error_id"}, nil)
+	if err != nil {
+		return false
+	}
+	return "CS.20005" == convertToStr(c)
```
Comment (Lance52259) [inline huaweicloud/services/gaussdb/transport.go:198]: For the comparison of expressions, the left side tends to change, and the right side tends to unchange.

File: huaweicloud/services/gaussdb/transport.go [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+}
+
+func checkCsPeeringConnectV1DeleteFinished(data interface{}) bool {
+	c, err := navigateValue(data, []string{"error_id"}, nil)
+	if err != nil {
+		return false
+	}
+	return "CS.22103" == convertToStr(c)
```
Comment (Lance52259) [inline huaweicloud/services/gaussdb/transport.go:206]: ditto

**Comment by niuzhenguo** [comment]
it will not be included in this milestone, will update later

**Pull request closed without merging**